sdk/identity/azure-identity/src/client_secret_credential.cpp

```diff
@@ -45,7 +51,7 @@ ClientSecretCredential::ClientSecretCredential(
 ClientSecretCredential::ClientSecretCredential(
     std::string tenantId,
     std::string clientId,
-    std::string clientSecret,
+    std::string const& clientSecret,
```

Comment:
Good catch. Leaking client secrets on the heap is probably a bad thing(tm) :).

Reply:
The reason I did that is that we don't keep a copy of `clientSecret` so copy is not needed - we partially initialize `m_requestBody` and put it there. For the rest of the arguments, we now do keep a copy, so we can take them byval and `std::move()` them, and if they happen to be rvalues, it would be more efficient. If not - either way a copy needs to be created, be it for the argument or later during the `m_*` initialization.

Reply:
The thing about client secrets is that they're secrets. If we store them in a std::string (even temporarily) without clearing them after they're used, then we potentially leak those into memory, allowing an attacker to access them (see also the CLR's `SecureString` class).

In practice, it's not a huge deal (and `SecureString` has a host of problems that are nasty enough that Barry Dorrans breaks out in hives whenever anyone mentions the phrase). But marking it as const is a good thing (since it means we don't need to copy the value on the call).